File: quiz-engine-springboot/src/main/java/com/quizengine/entity/QuizSession.java
package com.quizengine.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "quiz_sessions", indexes = {
    @Index(name = "idx_started_date", columnList = "started_at")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizSession {

    @Id
    @Column(length = 36)
    private String sessionId;

    @Column(nullable = false)
    @CreationTimestamp
    private LocalDateTime startedAt;

    @Column
    private LocalDateTime endedAt;

    @Column(nullable = false)
    private Integer numQuestions;

    @Column
    @Builder.Default
    private Integer numCorrect = 0;

    @Column
    @Builder.Default
    private Double percentageCorrect = 0.0;

    @Column
    private Integer timeTakenSeconds;

    @OneToMany(mappedBy = "session", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<QuizResponse> responses;
}

File: quiz-nodejs-engine/src/models/QuizSession.ts
import 'reflect-metadata';
import {
  Entity,
  PrimaryColumn,
  Column,
  CreateDateColumn,
  OneToMany,
} from 'typeorm';
import { QuizResponse } from './QuizResponse';

@Entity('quiz_sessions')
export class QuizSession {
  @PrimaryColumn({ type: 'varchar', length: 36 })
  sessionId!: string;

  @CreateDateColumn()
  startedAt!: Date;

  @Column({ type: 'datetime', nullable: true })
  endedAt?: Date;

  @Column({ type: 'integer' })
  numQuestions!: number;

  @Column({ type: 'integer', default: 0 })
  numCorrect!: number;

  @Column({ type: 'decimal', precision: 5, scale: 2, default: 0 })
  percentageCorrect!: number;

  @Column({ type: 'integer', nullable: true })
  timeTakenSeconds?: number;

  @OneToMany(() => QuizResponse, (response) => response.session, { lazy: true })
  responses!: Promise<QuizResponse[]>;
}
